posthog/test/__snapshots__/test_feature_flag.ambr

```diff
@@ -1,10 +1,121 @@
+# name: TestFeatureFlagHashKeyOverrides.test_entire_flow_with_hash_key_override
+  '
+  SELECT "posthog_featureflag"."id",
+         "posthog_featureflag"."key",
+         "posthog_featureflag"."filters",
+         "posthog_featureflag"."rollout_percentage",
+         "posthog_featureflag"."team_id",
+         "posthog_featureflag"."ensure_experience_continuity"
+  FROM "posthog_featureflag"
+  WHERE ("posthog_featureflag"."active"
+         AND NOT "posthog_featureflag"."deleted"
+         AND "posthog_featureflag"."team_id" = 2)
+  '
+---
+# name: TestFeatureFlagHashKeyOverrides.test_entire_flow_with_hash_key_override.1
+  '
+  SELECT "posthog_persondistinctid"."person_id"
+  FROM "posthog_persondistinctid"
+  WHERE ("posthog_persondistinctid"."distinct_id" = 'other_id'
+         AND "posthog_persondistinctid"."team_id" = 2)
+  LIMIT 1
+  '
+---
+# name: TestFeatureFlagHashKeyOverrides.test_entire_flow_with_hash_key_override.2
+  '
+  SELECT "posthog_persondistinctid"."person_id"
+  FROM "posthog_persondistinctid"
+  WHERE ("posthog_persondistinctid"."distinct_id" = 'example_id'
+         AND "posthog_persondistinctid"."team_id" = 2)
+  LIMIT 1
+  '
+---
+# name: TestFeatureFlagHashKeyOverrides.test_entire_flow_with_hash_key_override.3
+  '
+  SELECT "posthog_featureflaghashkeyoverride"."id",
+         "posthog_featureflaghashkeyoverride"."feature_flag_key"
+  FROM "posthog_featureflaghashkeyoverride"
+  WHERE ("posthog_featureflaghashkeyoverride"."person_id" = 2
+         AND "posthog_featureflaghashkeyoverride"."team_id" = 2)
+  '
+---
+# name: TestFeatureFlagHashKeyOverrides.test_entire_flow_with_hash_key_override.4
+  '
+  INSERT INTO "posthog_featureflaghashkeyoverride" ("feature_flag_key",
+                                                    "person_id",
+                                                    "team_id",
+                                                    "hash_key")
+  VALUES ('beta-feature', 1374, 386, 'example_id'),
+         ('multivariate-flag', 1374, 386, 'example_id') RETURNING "posthog_featureflaghashkeyoverride"."id"
+  '
+---
+# name: TestFeatureFlagHashKeyOverrides.test_entire_flow_with_hash_key_override.5
+  '
+  SELECT "posthog_featureflaghashkeyoverride"."feature_flag_key",
+         "posthog_featureflaghashkeyoverride"."hash_key"
+  FROM "posthog_featureflaghashkeyoverride"
+  WHERE ("posthog_featureflaghashkeyoverride"."person_id" = 2
+         AND "posthog_featureflaghashkeyoverride"."team_id" = 2)
+  '
+---
+# name: TestFeatureFlagHashKeyOverrides.test_entire_flow_with_hash_key_override.6
+  '
+  SELECT "posthog_featureflagoverride"."id",
+         "posthog_featureflagoverride"."feature_flag_id",
+         "posthog_featureflagoverride"."override_value",
+         "posthog_featureflag"."id",
+         "posthog_featureflag"."key"
+  FROM "posthog_featureflagoverride"
```

Comment:
this cruft from existing overrides makes me sad.

Specially, since this applies to a very small minority of users: Those who have a posthog account, while being called for every user coming to the client's website.

At minimum, we should just check if this distinct ID belongs to a posthog user, and then use the value inside this query. Hmm, but then, if featureflagoverride table is small, not sure it makes a difference.

Reply:
Could we kill this then? This won't affect any real operations and we could if we wanted to.

Reply:
Good question. I think killing this will remove the toolbar feature flags functionality. Don't think it's used anywhere else, is it? (cc: @mariusandra @rcmarron )

At the very least, I suppose we could go down this code path only when asking for `/my_flags` or `/decide` has a parameter used by the toolbar to make it happen.

This way, we can pretty much get the best of both worlds, by keeping this functionality, and removing cruft completely for most users.

Although I do question if the toolbar override functionality is used at all, @marcushyett-ph ?